aerogear/unifiedpush-operator
# Aerogear 9968

JIRAs:
* https://issues.jboss.org/browse/AEROGEAR-9966
* https://issues.jboss.org/browse/AEROGEAR-9968

~~Builds on top of https://github.com/aerogear/unifiedpush-operator/pull/73~~
~~Wait for that one to merge first~~
UPDATE: https://github.com/aerogear/unifiedpush-operator/pull/73 is closed and the code changes in that one will be merged within this PR

Changes:
* Reconciliation of resources based on static image reference. Done by deleting the things done during https://github.com/aerogear/unifiedpush-operator/pull/28 and reapplying things done during https://github.com/aerogear/unifiedpush-operator/pull/8
* Migrates to Deployments from DeploymentConfigs
* Deletes old resources of DeploymentConfigs and ImageStreams

Verification 1:
* Prepare the cluster, start the operator, install the CR --> UPS is running
* Check if the happy case is working fine (play with UPS console)
* Verify the SOP

Verification 2:
* Change the image in one of the deployments
* You will see the image is reverted to what's defined in the operator

Verification 3:
* Install an old version of the UPS operator and create a UPS instance with it
* Upgrade the operator to the PR version
  * Change the image in `deploy/operator.yaml` to `docker.io/aliok/upsoproct81655`
  * Run `cluster/prepare`
  * Run `oc apply -f deploy/role.yaml`
  * Run `oc apply -f deploy/operator.yaml`
* DeploymentConfigs and ImageStreams should be deleted
* Check if the happy case is working fine (play with UPS console)
* Existing push apps, variants etc. should be there, even though they're created over the UPS console and not by CRs (my purpose: check if switching from Postgres DC to Deployment worked fine and it still uses the same PV and the data is not gone)
